warehouse/annotation-core/src/main/java/datawave/annotation/util/Validator.java

```diff
@@ -0,0 +1,112 @@
+package datawave.annotation.util;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
+
+import datawave.data.hash.UID;
+
+/**
+ * A class for defining logic the object validator pattern, rules are added by passing predicates to {@link #addCheck(Predicate, String)} and object instances
+ * are validated using {@link #check(Object)}.
+ * <p>
+ * validators should be created once and reused. state is externalized in {@link ValidationState}, so it is safe to use a single instance across multiple
+ * threads, but initialization is not thread-safe.
+ * </p>
+ *
+ * @param <T>
+ *            the type of object that will be validated.
+ */
+public class Validator<T> {
+
+    /** the predicates to use for validation */
+    final Map<Predicate<T>,String> predicates = new HashMap<>();
+
+    /**
+     * create a validator, type is inferred through assignment or cast, e.g.:
+     *
+     * <pre>
+     * Validator<TestObject> validator = Validator.create();
+     * </pre>
+     *
+     * When chaining, use the method return cast:
+     *
+     * <pre>
+     *      Validator<TestObject> validator = Validator.<TestObject>create().addCheck(o -> o.name != null, "name is not null");
+     * </pre>
+     *
+     * @return a validator
+     * @param <V>
+     *            the type of object that will be validated
+     */
+    public static <V> Validator<V> create() {
+        return new Validator<>();
+    }
+
+    /** protected constructor use {@link #create()} to create a validator instance. */
+    protected Validator() {
+
+    }
+
+    /**
+     * add a predicate for validating objects of type T
+     *
+     * @param predicate
+     *            the predicate to execute
+     * @param errorMessage
+     *            the error message to store if validation fails
+     * @return this validator for chaining rule creation.
+     */
+    public Validator<T> addCheck(Predicate<T> predicate, String errorMessage) {
+        predicates.put(predicate, errorMessage);
+        return this;
+    }
+
+    public ValidationState<T> check(T target) {
+        ValidationState<T> state = new ValidationState<>(target);
+        for (Map.Entry<Predicate<T>,String> pe : predicates.entrySet()) {
+            if (!pe.getKey().test(target)) {
+                state.addError(pe.getValue());
```

Comment:
Do you want to test all predicates or only test predicates until a failure occurs?